### DISCUSSION TITLE
VcClusterComputeResource type dilemma

### DISCUSSION THREAD

**Discussion by unbreakabl3** [vmware/build-tools-for-vmware-aria#299] [General]
Hi all.
In vRO API, the "VcClusterComputeResource" is an Interface (in vRBT as well). But technically, it is a class. And it's acting as a class. Because it has an instance, once initialized/returned from vCenter. In JS, of course, this error doesn't exist. In my example, @ts-ignore solves the problem, and everything works once it is transpiled. Should the "VcClusterComputeResource" be a class in vRBT?

@Michaelpalacce , @joroaf - may you guys know?

![image](https://github.com/vmware/build-tools-for-vmware-aria/assets/13556249/9c895baa-e21a-42a4-9b97-1d6e5fa4c310)

**Comment by unbreakabl3** [comment]
Anyone?

**Comment by Michaelpalacce** [comment]
Hey @unbreakabl3 it should, yes... actually quite a lot of classes need to be changed to being classes. Writing a task for that now

**Reply by unbreakabl3** [reply]
I think I can try to work on that any soon and create a PR, so you can review it.

**Comment by VenelinBakalov** [comment]
Hi @unbreakabl3 @Michaelpalacce I just noticed this is still open. Did we update the code or it is still a pending issue?

**Reply by unbreakabl3** [reply]
Hi @VenelinBakalov ,
Sorry for the delay. I was quite busy. I will start to work on that as soon as possible.

**Comment by VenelinBakalov** [comment]
@unbreakabl3 no worries, just checking the status. Is it only changing from interface to class? Maybe I can try to close it before the current release happens?

**Comment by unbreakabl3** [comment]
Yes, it should be just a change from interface to class.

**Comment by VenelinBakalov** [comment]
Moved to an issue, I am currently working to close it, hopefully today.
https://github.com/vmware/build-tools-for-vmware-aria/issues/458

**Comment by unbreakabl3** [comment]
Hi @VenelinBakalov,
There are plenty of other interfaces like that. Do you want me to open another issue later on, so I can fix them slowly?

**Comment by VenelinBakalov** [comment]
if it is for vCenter plugin no need, I added them all to the PR, if it is for other plugins, let's add separate issues for each plugin
